Scripts/RemediationScripts/Get-ManagementCertificates.ps1

```diff
@@ -0,0 +1,352 @@
+        
+        .PARAMETER SubscriptionId
+        Specifies the ID of the Subscription to be remediated.
+        
+        .Parameter PerformPreReqCheck
+        Specifies validation of prerequisites for the command.
+
+        .INPUTS
+        None.
+
+        .OUTPUTS
+        None. Does not return anything that can be piped and used as an input to another command.
+
+        .EXAMPLE
+        PS> Get-ManagementCertificates -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -DryRun
+
+        .EXAMPLE
+        PS> Get-ManagementCertificates -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck
+
+        .EXAMPLE (If modules are already installed)
+        PS> Get-ManagementCertificates -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000
+
+        .LINK
+        None
+    #>
+
+      param (
+        [String]
+        [Parameter(ParameterSetName = "DryRun", Mandatory = $true, HelpMessage="Specifies the ID of the Subscription to be remediated")]
+        $SubscriptionId,
+
+        [Switch]
+        [Parameter(ParameterSetName = "DryRun", HelpMessage="Specifies validation of prerequisites for the command")]
+        $PerformPreReqCheck
+        )
+
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 1 of 2] Prepare to get management certificates in Subscription: [$($SubscriptionId)]"
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ($PerformPreReqCheck)
+    {
+        try
+        {
+            Write-Host "Setting up prerequisites..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            Setup-Prerequisites
+            Write-Host "Completed setting up prerequisites." -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::SingleDashLine)
+        }
+        catch
+        {
+            Write-Host "Error occurred while setting up prerequisites. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+            return
+        }
+    }
+    
+    
+     $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+
+     Write-Host "Subscription Name: [$($context.Subscription.Name)]"
+     Write-Host "Subscription ID: [$($context.Subscription.SubscriptionId)]"
+     Write-Host "Account Name: [$($context.Account.Id)]"
+     Write-Host "Account Type: [$($context.Account.Type)]"
+     Write-Host $([Constants]::SingleDashLine)
+    
+    # To get the Resource URL from environment
+    $ResourceAppIdURI = GetServiceManagementUrl
+
+    # Using Get-AzAccessToken to get access token for fetching management certificates.
+    $ClassicAccessToken= (Get-AzAccessToken -ResourceUrl $ResourceAppIdURI).Token
+    
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 2] Fetching  management certificates in Subscription: [$($SubscriptionId)]"
+    Write-Host $([Constants]::SingleDashLine)
+
+    if($null -ne $ClassicAccessToken)
+    {
```

Comment:
Just log message there was issue while fetching mgt cert in this case as we can't proceed w/o access token.

Reply:
Added.